### ISSUE TITLE
Manager button missing - errors in startup

### ISSUE THREAD

**Issue by alenknight** [Comfy-Org/ComfyUI-Manager#112]
had to reinstall comfyui... getting this error - particularly when i try to load ComfyUI-Manager-and the manager button is missing in my GUI
i've tried loading other nodes... and they are missing too.
```
C:\AI\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable>.\python_embeded\python.exe -s ComfyUI\main.py --windows-standalone-build
** ComfyUI start up time: 2023-09-20 11:48:05.405891

Prestartup times for custom nodes:
   0.0 seconds: C:\AI\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager

Total VRAM 24576 MB, total RAM 32704 MB
xformers version: 0.0.21
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3090 : cudaMallocAsync
VAE dtype: torch.bfloat16
Using xformers cross attention
Adding extra search path checkpoints C:\AI\stable-diffusion-webui\models/Stable-diffusion
Adding extra search path configs C:\AI\stable-diffusion-webui\models/Stable-diffusion
Adding extra search path vae C:\AI\stable-diffusion-webui\models/VAE
Adding extra search path loras C:\AI\stable-diffusion-webui\models/Lora
Adding extra search path loras C:\AI\stable-diffusion-webui\models/LyCORIS
Adding extra search path upscale_models C:\AI\stable-diffusion-webui\models/ESRGAN
Adding extra search path upscale_models C:\AI\stable-diffusion-webui\models/RealESRGAN
Adding extra search path upscale_models C:\AI\stable-diffusion-webui\models/SwinIR
Adding extra search path embeddings C:\AI\stable-diffusion-webui\embeddings
Adding extra search path hypernetworks C:\AI\stable-diffusion-webui\models/hypernetworks
Adding extra search path controlnet C:\AI\stable-diffusion-webui\models/ControlNet
## ComfyUI-Manager: installing dependencies
Traceback (most recent call last):
  File "C:\AI\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\__init__.py", line 30, in <module>
    import git
ModuleNotFoundError: No module named 'git'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\AI\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1725, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "C:\AI\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\__init__.py", line 37, in <module>
    run_script([sys.executable, '-s', '-m', 'pip', 'install', '-r', requirements_path])
  File "C:\AI\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\__init__.py", line 15, in run_script
    process = subprocess.Popen(cmd, cwd=cwd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True, bufsize=1)
NameError: name 'subprocess' is not defined

Cannot import C:\AI\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager module for custom nodes: name 'subprocess' is not defined

Import times for custom nodes:
   0.0 seconds (IMPORT FAILED): C:\AI\ComfyUI_windows_portable_nvidia_cu118_or_cpu\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager

Starting server

To see the GUI go to: http://127.0.0.1:8188
```

**Comment by alenknight** [comment]
and i have 'git'... I use git pull all the time....

**Comment by CxR0b505** [comment]
Use installation method #2 instead and it should work.

**Comment by alenknight** [comment]
> Use installation method #2 instead and it should work.

do you use this radial?
![image](https://github.com/ltdrdata/ComfyUI-Manager/assets/22153900/c27f1194-923b-48bd-898c-e454a119b5a7)

**Comment by alenknight** [comment]
looks like config.ini isn't being auto-created.  any idea why?

**Comment by ccarstens** [comment]
I have the same issue and I'm on linux so I can't use installation option 2 ://

**Comment by pippaine** [comment]
Experiencing same issue on linux + installation method 1

**Comment by Dkdnzia** [comment]
Exact same error here. Used installation method 2 without any luck, also config.ini isn't generating either. I got ComfyUI in a symlink. could that be the problem? Here is my startup log:


launching ComfyUI at C:\Users\david\seait\ComfyUI
** ComfyUI start up time: 2023-09-21 00:41:31.358516

Prestartup times for custom nodes:
   0.0 seconds: D:\SEAIT\ComfyUI\custom_nodes\ComfyUI-Manager

Total VRAM 24563 MB, total RAM 32686 MB
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 4090 :
VAE dtype: torch.float32
Using sub quadratic optimization for cross attention, if you have memory or speed issues try using: --use-split-cross-attention
Adding extra search path checkpoints D:\SEAIT\stable-diffusion-webui\models/Stable-diffusion
Adding extra search path configs D:\SEAIT\stable-diffusion-webui\models/Stable-diffusion
Adding extra search path vae D:\SEAIT\stable-diffusion-webui\models/VAE
Adding extra search path loras D:\SEAIT\stable-diffusion-webui\models/Lora
Adding extra search path loras D:\SEAIT\stable-diffusion-webui\models/LyCORIS
Adding extra search path upscale_models D:\SEAIT\stable-diffusion-webui\models/ESRGAN
Adding extra search path upscale_models D:\SEAIT\stable-diffusion-webui\models/RealESRGAN
Adding extra search path upscale_models D:\SEAIT\stable-diffusion-webui\models/SwinIR
Adding extra search path embeddings D:\SEAIT\stable-diffusion-webui\embeddings
Adding extra search path hypernetworks D:\SEAIT\stable-diffusion-webui\models/hypernetworks
Adding extra search path controlnet D:\SEAIT\stable-diffusion-webui\models/ControlNet
## ComfyUI-Manager: installing dependencies
Traceback (most recent call last):
  File "D:\SEAIT\ComfyUI\custom_nodes\ComfyUI-Manager\__init__.py", line 30, in <module>
    import git
ModuleNotFoundError: No module named 'git'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\david\seait\ComfyUI\nodes.py", line 1725, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\SEAIT\ComfyUI\custom_nodes\ComfyUI-Manager\__init__.py", line 37, in <module>
    run_script([sys.executable, '-s', '-m', 'pip', 'install', '-r', requirements_path])
  File "D:\SEAIT\ComfyUI\custom_nodes\ComfyUI-Manager\__init__.py", line 15, in run_script
    process = subprocess.Popen(cmd, cwd=cwd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True, bufsize=1)
NameError: name 'subprocess' is not defined

Cannot import D:\SEAIT\ComfyUI\custom_nodes\ComfyUI-Manager module for custom nodes: name 'subprocess' is not defined

Import times for custom nodes:
   0.0 seconds (IMPORT FAILED): D:\SEAIT\ComfyUI\custom_nodes\ComfyUI-Manager

Starting server

To see the GUI go to: http://127.0.0.1:8188

**Comment by ccarstens** [comment]
The issue seems to have been introduced after `3edb3586b7077f9d6d495d48caa79b20d1572ece`.

When you checkout that commit with 
```
git checkout 3edb3586b7077f9d6d495d48caa79b20d1572ece
```

and then restart ComfyUI I see the manager button.

Not sure how safe that is though, so waiting for a fix might be a good idea :)

**Comment by ltdrdata** [comment]
OMG... How this happen!!
I'd just fixed... check it.

**Comment by ltdrdata** [comment]
> The issue seems to have been introduced after `3edb3586b7077f9d6d495d48caa79b20d1572ece`.
> 
> When you checkout that commit with 
> ```
> git checkout 3edb3586b7077f9d6d495d48caa79b20d1572ece
> ```
> 
> and then restart ComfyUI I see the manager button.
> 
> Not sure how safe that is though, so waiting for a fix might be a good idea :)

If someone did this.

git checkout main
git pull

**Comment by Dkdnzia** [comment]
Works now, THANKS A LOT!!!!

**Comment by ccarstens** [comment]
thank you @ltdrdata ! <3 <3

**Comment by alenknight** [comment]
> > The issue seems to have been introduced after `3edb3586b7077f9d6d495d48caa79b20d1572ece`.
> > When you checkout that commit with
> > ```
> > git checkout 3edb3586b7077f9d6d495d48caa79b20d1572ece
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and then restart ComfyUI I see the manager button.
> > Not sure how safe that is though, so waiting for a fix might be a good idea :)
> 
> If someone did this.
> 
> git checkout main git pull

going back to main isn't working for me.... 3edb3586b7077f9d6d495d48caa79b20d1572ece works.... i can use that for now.....but yeah if i do the "git checkout main"... it breaks it again....

**Comment by ltdrdata** [comment]
> > > The issue seems to have been introduced after `3edb3586b7077f9d6d495d48caa79b20d1572ece`.
> > > When you checkout that commit with
> > > ```
> > > git checkout 3edb3586b7077f9d6d495d48caa79b20d1572ece
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > and then restart ComfyUI I see the manager button.
> > > Not sure how safe that is though, so waiting for a fix might be a good idea :)
> > 
> > 
> > If someone did this.
> > git checkout main git pull
> 
> going back to main isn't working for me.... [3edb358](https://github.com/ltdrdata/ComfyUI-Manager/commit/3edb3586b7077f9d6d495d48caa79b20d1572ece) works.... i can use that for now.....but yeah if i do the "git checkout main"... it breaks it again....

You should `git pull` after `git checkout main`

**Comment by alenknight** [comment]
yep.  i am.  even tried delete, reinstall.  still same error.  (i'm on the latest build of git too).
here's my entire terminal from cmd 
```
Microsoft Windows [Version 10.0.19045.3448]
(c) Microsoft Corporation. All rights reserved.

C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager>git checkout main
Previous HEAD position was 3edb358 update DB
Switched to branch 'main'
Your branch is up to date with 'origin/main'.

C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager>git pull
Already up to date.

C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager>cd..

C:\AI\ComfyUI\ComfyUI\custom_nodes>cd..

C:\AI\ComfyUI\ComfyUI>cd..

C:\AI\ComfyUI>run_nvidia_gpu.bat

C:\AI\ComfyUI>.\python_embeded\python.exe -s ComfyUI\main.py --listen --windows-standalone-build
** ComfyUI start up time: 2023-09-20 21:12:45.101021

Prestartup times for custom nodes:
   0.0 seconds: C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager

Total VRAM 24576 MB, total RAM 32704 MB
xformers version: 0.0.21
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3090 : cudaMallocAsync
VAE dtype: torch.bfloat16
Using xformers cross attention
### Loading: ComfyUI-Impact-Pack (V4.8.3)
### Loading: ComfyUI-Impact-Pack (Subpack: V0.2.2)
### Loading: ComfyUI-Manager (V0.30.2)
### ComfyUI Revision: 1457 [0665749b]

Import times for custom nodes:
   0.3 seconds: C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager
   0.6 seconds: C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Impact-Pack

Starting server

To see the GUI go to: http://0.0.0.0:8188
FETCH DATA from: C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager\extension-node-map.json
FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json
Install custom node 'ComfyUI Inspire Pack'
install: ['https://github.com/ltdrdata/ComfyUI-Inspire-Pack']
Download: git clone 'https://github.com/ltdrdata/ComfyUI-Inspire-Pack'
[!] Traceback (most recent call last):
[!]   File "C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager\git_helper.py", line 86, in <module>
[!]     setup_environment()
[!]   File "C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager\git_helper.py", line 82, in setup_environment
[!]     if 'git_exe' in config['default'] and config['default']['git_exe'] != '':
[!]   File "configparser.py", line 964, in __getitem__
[!] KeyError: 'default'
Installation was successful.
After restarting ComfyUI, please refresh the browser.
FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json
```

**Comment by ltdrdata** [comment]
> yep. i am. even tried delete, reinstall. still same error. (i'm on the latest build of git too). here's my entire terminal from cmd
> 
> ```
> Microsoft Windows [Version 10.0.19045.3448]
> (c) Microsoft Corporation. All rights reserved.
> 
> C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager>git checkout main
> Previous HEAD position was 3edb358 update DB
> Switched to branch 'main'
> Your branch is up to date with 'origin/main'.
> 
> C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager>git pull
> Already up to date.
> 
> C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager>cd..
> 
> C:\AI\ComfyUI\ComfyUI\custom_nodes>cd..
> 
> C:\AI\ComfyUI\ComfyUI>cd..
> 
> C:\AI\ComfyUI>run_nvidia_gpu.bat
> 
> C:\AI\ComfyUI>.\python_embeded\python.exe -s ComfyUI\main.py --listen --windows-standalone-build
> ** ComfyUI start up time: 2023-09-20 21:12:45.101021
> 
> Prestartup times for custom nodes:
>    0.0 seconds: C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager
> 
> Total VRAM 24576 MB, total RAM 32704 MB
> xformers version: 0.0.21
> Set vram state to: NORMAL_VRAM
> Device: cuda:0 NVIDIA GeForce RTX 3090 : cudaMallocAsync
> VAE dtype: torch.bfloat16
> Using xformers cross attention
> ### Loading: ComfyUI-Impact-Pack (V4.8.3)
> ### Loading: ComfyUI-Impact-Pack (Subpack: V0.2.2)
> ### Loading: ComfyUI-Manager (V0.30.2)
> ### ComfyUI Revision: 1457 [0665749b]
> 
> Import times for custom nodes:
>    0.3 seconds: C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager
>    0.6 seconds: C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Impact-Pack
> 
> Starting server
> 
> To see the GUI go to: http://0.0.0.0:8188
> FETCH DATA from: C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager\extension-node-map.json
> FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json
> Install custom node 'ComfyUI Inspire Pack'
> install: ['https://github.com/ltdrdata/ComfyUI-Inspire-Pack']
> Download: git clone 'https://github.com/ltdrdata/ComfyUI-Inspire-Pack'
> [!] Traceback (most recent call last):
> [!]   File "C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager\git_helper.py", line 86, in <module>
> [!]     setup_environment()
> [!]   File "C:\AI\ComfyUI\ComfyUI\custom_nodes\ComfyUI-Manager\git_helper.py", line 82, in setup_environment
> [!]     if 'git_exe' in config['default'] and config['default']['git_exe'] != '':
> [!]   File "configparser.py", line 964, in __getitem__
> [!] KeyError: 'default'
> Installation was successful.
> After restarting ComfyUI, please refresh the browser.
> FETCH DATA from: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json
> ```

Oh... that is another fault... wait a minute.

**Comment by ltdrdata** [comment]
Try again... Just patched...

**Comment by alenknight** [comment]
> Try again... Just patched...

NICE!  now it worked!  what was it?

thanks again!!  awesome!

**Comment by ltdrdata** [comment]
> > Try again... Just patched...
> 
> NICE! now it worked! what was it?
> 
> thanks again!! awesome!

I recently added a feature to the config, but it caused issues when the config file doesn't exist.

**Comment by UnclePsymon** [comment]
Does not work here on Linux with a fresh installation

```
root@nixos /d/ComfyUI (master)# HSA_OVERRIDE_GFX_VERSION=10.3.0 python main.py
** ComfyUI start up time: 2023-11-05 19:46:46.847438

Prestartup times for custom nodes:
   0.0 seconds: /dockerx/ComfyUI/custom_nodes/ComfyUI-Manager

Total VRAM 12272 MB, total RAM 32006 MB
Set vram state to: NORMAL_VRAM
Device: cuda:0 AMD Radeon RX 6700 XT : native
VAE dtype: torch.float32
Using sub quadratic optimization for cross attention, if you have memory or speed issues try using: --use-split-cross-attention
### Loading: ComfyUI-Manager (V0.39.1)
## ComfyUI-Manager: installing dependencies
Traceback (most recent call last):
  File "/dockerx/ComfyUI/custom_nodes/ComfyUI-Manager/__init__.py", line 39, in <module>
    import git
ModuleNotFoundError: No module named 'git'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/dockerx/ComfyUI/nodes.py", line 1735, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/dockerx/ComfyUI/custom_nodes/ComfyUI-Manager/__init__.py", line 46, in <module>
    run_script([sys.executable, '-s', '-m', 'pip', 'install', '-r', requirements_path])
  File "/dockerx/ComfyUI/custom_nodes/ComfyUI-Manager/__init__.py", line 24, in run_script
    process = subprocess.Popen(cmd, cwd=cwd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True, bufsize=1)
NameError: name 'subprocess' is not defined

Cannot import /dockerx/ComfyUI/custom_nodes/ComfyUI-Manager module for custom nodes: name 'subprocess' is not defined

Import times for custom nodes:
   0.0 seconds (IMPORT FAILED): /dockerx/ComfyUI/custom_nodes/ComfyUI-Manager

Starting server

To see the GUI go to: http://127.0.0.1:8188
```

**Comment by ltdrdata** [comment]
> Does not work here on Linux with a fresh installation
> 
> ```
> root@nixos /d/ComfyUI (master)# HSA_OVERRIDE_GFX_VERSION=10.3.0 python main.py
> ** ComfyUI start up time: 2023-11-05 19:46:46.847438
> 
> Prestartup times for custom nodes:
>    0.0 seconds: /dockerx/ComfyUI/custom_nodes/ComfyUI-Manager
> 
> Total VRAM 12272 MB, total RAM 32006 MB
> Set vram state to: NORMAL_VRAM
> Device: cuda:0 AMD Radeon RX 6700 XT : native
> VAE dtype: torch.float32
> Using sub quadratic optimization for cross attention, if you have memory or speed issues try using: --use-split-cross-attention
> ### Loading: ComfyUI-Manager (V0.39.1)
> ## ComfyUI-Manager: installing dependencies
> Traceback (most recent call last):
>   File "/dockerx/ComfyUI/custom_nodes/ComfyUI-Manager/__init__.py", line 39, in <module>
>     import git
> ModuleNotFoundError: No module named 'git'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/dockerx/ComfyUI/nodes.py", line 1735, in load_custom_node
>     module_spec.loader.exec_module(module)
>   File "<frozen importlib._bootstrap_external>", line 850, in exec_module
>   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
>   File "/dockerx/ComfyUI/custom_nodes/ComfyUI-Manager/__init__.py", line 46, in <module>
>     run_script([sys.executable, '-s', '-m', 'pip', 'install', '-r', requirements_path])
>   File "/dockerx/ComfyUI/custom_nodes/ComfyUI-Manager/__init__.py", line 24, in run_script
>     process = subprocess.Popen(cmd, cwd=cwd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True, bufsize=1)
> NameError: name 'subprocess' is not defined
> 
> Cannot import /dockerx/ComfyUI/custom_nodes/ComfyUI-Manager module for custom nodes: name 'subprocess' is not defined
> 
> Import times for custom nodes:
>    0.0 seconds (IMPORT FAILED): /dockerx/ComfyUI/custom_nodes/ComfyUI-Manager
> 
> Starting server
> 
> To see the GUI go to: http://127.0.0.1:8188
> ```

This is fixed on v0.39.2
